napari/plugins/hook_specifications.py

```diff
@@ -30,23 +30,87 @@
     ``napari_hook_specification`` and ``napari_hook_implementation``,
     respectively.
 """
-
 # These hook specifications also serve as the API reference for plugin
 # developers, so comprehensive documentation with complete type annotations is
 # imperative!
-
-from types import FunctionType
-from typing import Any, List, Optional, Union
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Union,
+)
 
 from napari_plugin_engine import napari_hook_specification
 
-from ..types import AugmentedWidget, ReaderFunction, WriterFunction
+if TYPE_CHECKING:
+    from pathlib import Path
+    from types import FunctionType
+
+    from ..types import (
+        AugmentedWidget,
+        LayerData,
+        ReaderFunction,
+        WriterFunction,
+    )
 
 # -------------------------------------------------------------------------- #
 #                                 IO Hooks                                   #
 # -------------------------------------------------------------------------- #
 
 
+@napari_hook_specification(historic=True)
```

Comment:
Do we need to update some `docs` or that will be done in a separate PR?

Reply:
This page is what generates the content for https://napari.org/plugins/stable/hook_specifications.html ... so this _is_ the docs... does that cover it?  or did you mean something more narrative?

Reply:
Covers it :)

Reply:
I have no idea how docs work 😆